tests/stdlib/tresults.nim

```diff
@@ -0,0 +1,209 @@
+discard """
+  description: "Tests for the std/experimental/results module"
+  matrix: "--gc:refc; --gc:arc"
+  targets: "c cpp js"
+"""
+
+# This file was based on the ``test.nim`` file
```

Comment:
Would it be asking too much to use `unittest` inside here?

I think a unit testing library is a better direction for stdlib unit testing imo, rather than mostly stylistic structures through blocks and the lik. <-- feel free to disagree

Reply:
> Would it be asking too much to use `unittest` inside here?

I don't think so, no. Unfortunately, `std/unittest` doesn't work at compile-time (yet), since it uses non-compile-time globals and methods.

I'm not against using `unittest` here, but it would mean that `Result` can't be tested in the VM for now.

Reply:
Oh, that's silly. Ok, looks like outside of this unit test needs to be fixed. Nevermind for now.